@here/harp-mapview/lib/MapView.ts

```diff
@@ -2438,13 +2464,14 @@ export class MapView extends THREE.EventDispatcher {
 
         const cameraPitch = MapViewUtils.extractAttitude(this, this.m_camera).pitch;
         const cameraPosZ = this.getCameraHeightAboveTerrain(TERRAIN_ZOOM_LEVEL);
+        const zoomLevelDistance = cameraPosZ / Math.cos(Math.min(cameraPitch, Math.PI / 3));
```

Comment:
I think this is all messed up. `zoomLevelDistance` should be the distance to the target point, but currently this is not true for globe and also if the tilt is > `60°`.

Reply:
Ohhh yes, that is all true, see also all my comments below (I started responding bottom-up).
- for globe we could improve the calculations the have exact distance to point on the ground (this it actually `lookAtDistance`.
- as for 60 deg constraint, I guess that it was provided to limit effect of vanishing (target) point, when zoom level dist goes to infinity when camera pitches close to 90 degrees.  
So having said all that here and below. What would you suggest?
I have a feeling that there are two possible approaches:
- we need to "cache" some of the params of the equation (last target/pivot point, or its last distance), when camera rotates and looking to the void,
- we could calculate "proper" target point on the ground (as it is now, but without raycasting) until the camera "ray" closes to horizon, then for globe it is limited by sphere tangent (furthest point visible) and for planar projections we should constraint it with some angle (89,9), because at 90 degrees our ray goes to infinity.
Do you have any other ideas?